b.c

```diff
@@ -931,10 +934,10 @@ bool fnematch(fa *pfa, FILE *f, char **pbuf, int *pbufsize, int quantum)
 		 * terminate the buffer.
 		 */
 		do
-			if (*--k && ungetc(*k, f) == EOF)
-				FATAL("unable to ungetc '%c'", *k);
-		while (k > patbeg + patlen);
```

Comment:
This is incorrect and causes failures in bugs-fixed/unicode-fs-rs-1.awk and bugs-fixed/unicode-fs-rs-2.awk.  It should be:
```
while (k + buf > patbeg + patlen);
```
or:
```
while (k > patbeg - buf + patlen);
```
since buf != patbeg

Reply:
Oh thanks! I'm current implementing `RT` like in gawk, and I was trying to figure out why my matches were messing up badly - this was the reason!

Reply:
I picked your second option: `while (k > patbeg - buf + patlen);`